### PR TITLE
Swap: use correct fee on spending cap screen

### LINKED ISSUE #21428: Swap fee shown on spending cap screen instead of approval fee
### Steps:
1. Recover user with non-approved ERC-20
2. Go to swap page
3. Navigate to the "Spending Cap" screen during the swap approval process.
 

### Actual Result:
The swap fee is shown on the "Spending Cap" screen, where the approval fee should be displayed.

https://github.com/user-attachments/assets/96c70bfe-0f31-4cd5-a727-0f2534bee6eb

### Expected Result:
- When a non-approved ERC-20 token is selected, the swap page displays the total fee (approval + swap).
- After the user taps the "Approve" button, the "spending cap" screen appears with a lower fee that only reflects the "approval" fee.
- Once the approval is completed, the user is navigated back to the swap page where the fee now reflects only the "swap" fee.

https://discord.com/channels/1210237582470807632/1232982322874159104/1295315205009576000

Image

For more details read from this https://discord.com/channels/1210237582470807632/1232982322874159104/1295310003455918102 message

### Logs:
Logs can be provided for further debugging.

### Env:
Nightly 14 Oct, 2024

### PR DESCRIPTION
**Pull request by vkjr**
fixes #21428

### Summary

Incorrect fee was used before, that is fixed.
Also minor refactoring done

### Review notes
`get-standard-fiat-format` was refactored, explanation in comment to the code

#### Platforms
- Android
- iOS

##### Functional
- wallet / transactions

status: ready

### CODE REVIEW

File: src/status_im/contexts/wallet/common/utils.cljs
Comment (vkjr) [inline src/status_im/contexts/wallet/common/utils.cljs:254]: This function was renamed because we don't have any "standard" fiat format) But naming it as "ui"-related makes it clear that result shouldnt be used in calculations.
Also for some reason we were passing here crypto value and checking if it includes "<" sign. We actually don't need this because we have a fiat value. So extra argument removed.

File: src/status_im/contexts/wallet/swap/set_spending_cap/view.cljs
Comment (vkjr) [inline src/status_im/contexts/wallet/swap/set_spending_cap/view.cljs]: This is an actual bugfix.

**Review by briansztamfater — APPROVED** [review]
LGTM

File: src/status_im/subs/wallet/swap.cljs
Comment (clauxx) [inline src/status_im/subs/wallet/swap.cljs]: extra space

**Review by smohamedjavid — APPROVED** [review]

File: src/status_im/subs/wallet/swap.cljs
Comment (ulisesmac) [inline src/status_im/subs/wallet/swap.cljs:306]: nit: Consider using `some` instead of `first` + `filter`

**Review by ulisesmac — APPROVED** [review]
Thank you so much for you PR @vkjr !!

**Review by clauxx — APPROVED** [review]

**Comment by VolodLytvynenko** [comment]
hi @vkjr thank you for PR.
On the swap page, it seems the fee is not being calculated as 'approval fee + swap fee.'

In the table below
image

When I enter a non-approved value (5.989 DAI) on mobile, the difference between non-approved and approved ERC-20 (5.98 DAI) is only about 90 cents . However the approval fee for this transaction is approximately $4. 
image


I perfomed four tests on both [Desktop](https://www.dropbox.com/scl/fi/iatqhmjmr94wmzx3cf56w/maxfe_desktop.mp4?rlkey=4d8tetzi2uamop0xi180c35pf&st=i5c3c3be&dl=0) and mobile to minimize the impact of fee amount changes. 

[On Desktop](https://www.dropbox.com/scl/fi/iatqhmjmr94wmzx3cf56w/maxfe_desktop.mp4?rlkey=4d8tetzi2uamop0xi180c35pf&st=i5c3c3be&dl=0), where it appears to work correctly, the non-approved value is around $4 less than the approved value. Could you take a look at this?

### Steps:
1. Go to swap
2. Enter non-approved value
3. Enter appoved value
4. Check fee between  non-approved and approved ERC20

### Actual result:
seems on the swap fee for non-approved in calculated incorrectly or calculated including only swap fee

https://github.com/user-attachments/assets/960956a9-ab5e-4a35-a88b-180bf18cfc0d

### Expected result:
On the swap fee for non-approved includes approval fee + swap fee

**Comment by vkjr** [comment]
@VolodLytvynenko, thanks for checking, will take a look

**Comment by vkjr** [comment]
@VolodLytvynenko, I've updated how we show the "max fee":
On swap setup screen `max fee` calculated as `(swap proposal fee + approval fee)` in case when approval needed, or just `swap proposal fee` when approval is not needed.
On swap confirmation screen `max fee` equals to `swap proposal fee`
Desktop and mobile now show similar results.

@briansztamfater, could you please also take a look?
I remade subscriptions to provide non-formatted values and apply formatting in the views to avoid duplication of whole subscriptions code.

**Comment by VolodLytvynenko** [comment]
Unfortunately one more issue is found, however it can be considered as a follow up

## ISSUE 2: The fee is not re-estimated when approving is performed

### Steps to Reproduce:
1. Navigate to the Swap Page with a non-approved ERC-20 token. (in my case it is shown as: Total: 26 USD
2. Approve the ERC-20 token.
3. Wait until the ERC-20 approval is successfully confirmed.

### Actual Result:
After approval, the user is returned to the Swap Page, but the fee estimation remains as approval + swap (e.g., 26 USD).

https://github.com/user-attachments/assets/40836273-1a91-4d1a-ba1d-4e1f4fba6f04

### Expected Result:
After the approval is confirmed, the fee estimation should update to reflect only the swap fee (e.g., 20 USD).

### Additional info:
However the fee will be reestimated if swap set up page will be reopened

**Comment by vkjr** [comment]
@VolodLytvynenko, I fixed the issue. (tested, now max fees is less after approved)

**Comment by VolodLytvynenko** [comment]
hi @vkjr thank you for fixes. No issues from my side. PR can be merged

**Pull request merged**
